include/envoy/server/options.h

```diff
@@ -86,10 +86,15 @@ class Options {
   virtual const envoy::config::bootstrap::v2::Bootstrap& configProto() const PURE;
 
   /**
-   * @return bool allow unknown fields in the configuration?
+   * @return bool allow unknown fields in the static configuration?
    */
   virtual bool allowUnknownFields() const PURE;
```

Comment:
nit: personally I would call this `allowUnknownStaticFields` and similarly `rejectUnknownDynamicFields` but up to you.

Reply:
Ack. I think what I'll do is have two new flags, `--allow-unknown-static-fields` and `--reject-unknown-dynamic-fields`, make `--allow-unknown-fields` a deprecated CLI alias for the former.